packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -72,11 +73,39 @@ import MAX_SIGNED_31_BIT_INT from './maxSigned31BitInt';
 let didWarnAboutBadClass;
 let didWarnAboutGetDerivedStateOnFunctionalComponent;
 let didWarnAboutStatelessRefs;
+let emptyContextThatWarnsOnAccess;
 
 if (__DEV__) {
   didWarnAboutBadClass = {};
   didWarnAboutGetDerivedStateOnFunctionalComponent = {};
   didWarnAboutStatelessRefs = {};
+
+  emptyContextThatWarnsOnAccess =
+    typeof Proxy === 'function'
+      ? new Proxy(emptyObject, {
+          get(target, name) {
+            const owner = ReactCurrentOwner.current;
+            let info = '';
+            if (owner !== null) {
+              const componentName = getComponentName(owner);
+              if (componentName != null) {
```

Comment:
Is the use of  `!=` intended here? If so, why? Thanks.

Reply:
Its checking whether the name is null or undefined. `!= null` or `== null` is a common shorthand for `foo !== undefined && foo !== null` since a double equal against `null` coerce `undefined` to `null` but nothing else.

Reply:
There might be a slight performance benefit to not do the shortcut, even with a code size penalty, because `foo != null` is basically equivalent to `foo !== document.all && foo !== null && foo !== undefined`.

Welcome to `document.all` :psyduck:

Reply:
This is a DEV-only code path that won't be called frequently in most cases, so small performance tweaks aren't super important.